sky/utils/common_utils.py

```diff
@@ -732,3 +732,13 @@ def hash_file(path: str, hash_alg: str) -> 'hashlib._Hash':
                 break
             file_hash.update(view[:size])
         return file_hash
+
+
+def is_port_available(port: int) -> bool:
+    """Check if the given port is available."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        try:
+            s.bind(('localhost', port))
+            return True
+        except socket.error:
+            return False
```

Comment:
Just to double check, it seems we are using `OSError` in the following place. Wondering is it a `socket.error` or a `OSError`.

https://github.com/aylei/skypilot/blob/ad97e8ef5a6d8cb6dea145d71c91002c79d48a16/sky/utils/common_utils.py#L529-L545

For the future, we can use the `find_free_port` function for the port used by queue manager : )

Reply:
https://docs.python.org/3/library/socket.html#socket.error socket error is an alias of OSError. Didn't notice the `find_free_port` function, I think it would a quick fix then, which also eliminates the extra knowledge we have to hold with ports